src/core/runQuery.ts

```diff
@@ -32,7 +32,19 @@ export interface QueryOptions {
  formatResponse?: Function;
 }
 
+// Make the global Promise constructor Fiber-aware.
+import { makeCompatible } from 'meteor-promise';
+import Fiber = require('fibers');
+makeCompatible(Promise, Fiber);
+
+const resolvedPromise = Promise.resolve();
+
 function runQuery(options: QueryOptions): Promise<GraphQLResult> {
+    // Fiber-aware Promises run their .then callbacks in Fibers.
+    return resolvedPromise.then(() => doRunQuery(options));
```

Comment:
Having this in apollo-server core is probably OK, because it's not specific to fibers.